### DISCUSSION TITLE
How to batch queries?

### DISCUSSION THREAD

**Discussion by arthot** [gajus/slonik#350] [Q&A]
I wonder if there is a way to run a batch of independent queries in the scope of one transaction? Something like this?

```js
const commands = [sql`insert into boo values (${id})`, sql`update table foo set "val" = ${val} where id = ${id}`];
await connection.transaction(async (t1) => {
  await t1.batch(commands);
});
```

The goal is to run them within one query, not one by one as it's way more efficient than sequential execution. The list `commands` is dynamic and can contain like a thousand commands. E.g. something like this won't work 

```js
await connection.transaction(async (t1) => {
  await t1.query(sql`insert into boo values (${id})`);
  await t1.query(sql`update table foo set "val" = ${val} where id = ${id}`);
});
```

According to the documentation, "Manually constructing queries is not allowed." so I cannot glue them manually. Any ideas?

**Comment by gajus** [comment]
This is a common topic in `node-postgres` issues https://github.com/brianc/node-postgres/issues/1388

9/10 there is a better way to write queries in a way that does not require batching. Other times, `Promise.all` will achieve equivalent performance.
